client/platform/desktop/backend/platforms/common.ts

```diff
@@ -202,58 +228,87 @@ async function createKwiverRunWorkingDir(datasetName: string, baseDir: string, p
+  const projectInfo = getProjectDir(settings, datasetId);
 
-  /* Move old file if it exists */
-  if (base.jsonFile) {
+  try {
+    const validatedInfo = await getValidatedProjectDir(settings, datasetId);
     await fs.move(
-      npath.join(base.basePath, base.jsonFile),
-      npath.join(auxFolderPath, base.jsonFile),
+      validatedInfo.trackFileAbsPath,
+      npath.join(
+        validatedInfo.auxDirAbsPath,
+        npath.basename(validatedInfo.trackFileAbsPath),
+      ),
     );
+  } catch (err) {
+    // Some part of the project dir didn't exist
   }
-
   const serialized = JSON.stringify(trackData);
-
-  /* Save new file */
-  await fs.writeFile(npath.join(base.basePath, newFileName), serialized);
+  await fs.writeFile(npath.join(projectInfo.basePath, newFileName), serialized);
 }
 
 /**
  * Save detections to json file in aux
- * @param datasetId path
- * @param args save args
  */
-async function saveDetections(datasetId: string, args: SaveDetectionsArgs) {
+async function saveDetections(settings: Settings, datasetId: string, args: SaveDetectionsArgs) {
   /* Update existing track file */
-  const existing = await loadDetections(datasetId, true);
+  const projectDirInfo = await getValidatedProjectDir(settings, datasetId);
+  const existing = await _loadJsonTracks(projectDirInfo.trackFileAbsPath);
   args.delete.forEach((trackId) => delete existing[trackId.toString()]);
   args.upsert.forEach((track, trackId) => {
     existing[trackId.toString()] = track.serialize();
   });
-  return saveSerialized(datasetId, existing);
+  return _saveSerialized(settings, datasetId, existing);
 }
 
 /**
- * Postprocess possible annotation files
- * @param paths paths to input annotation files in descending priority order.
- *              Only the first successful input will be loaded.
+ * _saveAsJson saves directly to disk
+ */
+async function _saveAsJson(absPath: string, data: unknown) {
+  const serialized = JSON.stringify(data, null, 2);
+  await fs.writeFile(absPath, serialized);
+}
+
+async function saveMetadata(settings: Settings, datasetId: string, args: DatasetMetaMutable) {
+  const projectDirInfo = await getValidatedProjectDir(settings, datasetId);
+  const existing = await loadMetadata(projectDirInfo.metaFileAbsPath);
+  if (args.confidenceFilters) {
+    existing.confidenceFilters = args.confidenceFilters;
+  }
+  if (args.customTypeStyling) {
+    existing.customTypeStyling = args.customTypeStyling;
+  }
+  _saveAsJson(projectDirInfo.metaFileAbsPath, existing);
+}
+
+/**
+ * processOtherAnnotationFiles imports data from external annotation formats
+ * given a list of candidate file paths.
+ *
+ * SUPPORTED FORMATS:
+ * VIAME CSV
+ *
+ * @param paths paths to possible input annotation files
  * @param datasetId dataset id path
  */
-async function postprocess(paths: string[], datasetId: string) {
-  for (let i = 0; i < paths.length; i += 1) {
-    const path = paths[i];
+async function processOtherAnnotationFiles(
```

Comment:
Another questions/Future thing can mark as resolved.  Just double checking that the intention is to do no cleanup after a job is done and leave the remnants (CSV files) within the  jobID folder?  We don't want to move the .csv files into the auxiliary folder for the Project to keep a sort of historical log of how the detentions/tracks changed?

Reply:
> We don't want to move the .csv files into the auxiliary folder for the Project to keep a sort of historical log of how the detentions/tracks changed?

This change record will be captured in the json auxiliary folder.  When a job results in detections, they're saved using the same save function as a user-initiated edit.  

I intended to leave the job dir alone so you could go in and look at what it leaves behind.